demos/LowAllocationWebServer/HttpHeaderBuffer.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Text;
+
+namespace System.Net.Http.Buffered
+{ 
+    public struct HttpHeaderBuffer
+    {        
+        private Span<byte> _valueByteSpan;
+
+        public HttpHeaderBuffer(Span<byte> valueByteSpan)
+        {
+            _valueByteSpan = valueByteSpan;
+        }
+
+        public void UpdateValue(string newValue)
+        {
+            if (newValue.Length > _valueByteSpan.Length)
+            {
+                throw new ArgumentException("newValue");
+            }
+
+            _valueByteSpan.Set(new UTF8Encoding().GetBytes(newValue));
```

Comment:
This allocates (encoding and byte array). You should use the formatting APIs to format directly into the span using the same "encoding" (FormattingData) that was used by the BufferFormatter.